### PR TITLE
Extend category definition

### PR DESCRIPTION
**Pull request by jethrolarson**

### CODE REVIEW

**Comment by jethrolarson** [comment]
I know this definition is getting kind of mathy. It kind of hard to talk about categories without a little bit.

File: readme.md [hunk 1]
```diff
@@ -324,9 +324,27 @@ addOne('some string') // Contract violated: expected int -> int
...
+    Where `a`, `b`, and `c` are objects in some category,
+    and `f` is a morphism from `a -> b`, and `g` is a morphism from `b -> c`;
+    `g(f(x))` must be equivalent to `(g • f)(x)`.
+3. Composition must be associative
+    `f • (g • h)` is the same as `(f • g) * h`
+
+Since these rules govern composition at very abstract level, category theory is great at uncovering new ways of composing things.
+
+### Further reading
+* [Category Theory for Programmers](https://bartoszmilewski.com/2014/10/28/category-theory-for-programmers-the-preface/)
```
Comment (hemanth) [inline readme.md]: I feel all the reference links or suggested reads we must put it to the wiki.

**Comment by hemanth** [comment]
Identity morphism, associativity and composition shouldn't be all that complex.

**Comment by jethrolarson** [comment]
Are you saying the definition is too complex?

**Comment by hemanth** [comment]
No, the other way round!

**Pull request merged**